api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -812,6 +821,14 @@ message Cluster {
   // of 0 would indicate that none of the timeout was used or that the timeout was infinite. A value
   // of 100 would indicate that the request took the entirety of the timeout given to it.
   bool track_timeout_budgets = 47;
+
+  // Optional customization and configuration of upstream type.
+  // By default, Envoy will create an upstream of type envoy.filters.upstreams.http.default
+  // The default upstream will create a envoy.filters.upstreams.http.tcp upstream
+  // if the request is a CONNECT request and connect termination is configured.
+  // Otherwise it will be envoy.filters.upstreams.http.http, with the protocol
+  // based on `http2_protocol_options`.
```

Comment:
I think this is still going to confuse people, so I think it might be worth spelling this out even more, especially the last part about being based on `http2_protocol_options`. This is already a point of huge confusion.